### PR TITLE
SITES-1162: Encrypt upgrade path

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Upgrade path handling for encrypt module when things go sidewise

# Needed By (Date)
- ?

# Urgency
- ?

# Steps to Test

1. Run upgrade path on jse-me
2. Validate all upgrade hooks finish

# Affected Projects or Products
- JSE and all CardinalD7?

# Associated Issues and/or People
- SITES-1162
- https://github.com/SU-SWS/acsf-cardinald7/pull/170

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: stanford_capx.install [hunk 1]
```diff
@@ -429,10 +429,19 @@ function stanford_capx_update_7301() {
...
-  variable_set('stanford_capx_password', encrypt($password));
+  try {
+    $username = decrypt(variable_get('stanford_capx_username', ''));
+    $password = decrypt(variable_get('stanford_capx_password', ''));
+    variable_set('stanford_capx_username', encrypt($username));
+    variable_set('stanford_capx_password', encrypt($password));
+  }
+  catch(Exception $e) {
+    // If decrypt and save fails remove the username and password to disable
+    // CAPx.
```
Comment (jbickar) [inline stanford_capx.install]: Log this if it happens?

Comment (sherakama) [reply stanford_capx.install]: encrypt already prints a message and adds a watchdog log.

Comment (sherakama) [reply stanford_capx.install]: https://github.com/SU-SWS/acsf-cardinald7/blob/1.x/docroot/sites/all/modules/contrib/encrypt/plugins/encryption_methods/mcrypt_aes_cbc.inc#L76

Comment (jbickar) [reply stanford_capx.install]: But does the user know that CAPx has been disabled because this happened?

Comment (sherakama) [reply stanford_capx.install]: Yeah, I suppose this could be better. Let me think about this a bit.

**Comment by sherakama** [comment]
@jbickar ready for review again. Order of operations change seems to have resolved the issues.

**Comment by sherakama** [comment]
POKE

**Comment by jbickar** [comment]
OK, I missed the note that this is ready for review again.

**Comment by jbickar** [comment]
So the encrypt upgrade path stuff already made it in, this is just bug-fixing via order-of-operations?

**Comment by jbickar** [comment]
Ah, right. This cropped up in https://github.com/SU-SWS/acsf-cardinald7/pull/170

**Comment by jbickar** [comment]
I decided to test this on a Lando box. I chose poorly.

**Comment by jbickar** [comment]
This works on `jse-me`.

**Review by jbickar — APPROVED** [review]
Tank yew.

**Pull request merged**